lib/AutomaticEmails/WooCommerce/Events/PurchasedProduct.php

```diff
@@ -118,9 +118,9 @@ function scheduleEmailWhenProductIsPurchased($order_id) {
     }
 
     $ordered_products = array_map(function($product) {
-      return $product->get_product_id();
+      return is_callable([$product, 'get_product_id']) ? $product->get_product_id() : null;
     }, $order_details->get_items());
-
+    $ordered_products = array_filter($ordered_products);
```

Comment:
I'd add `array_values` after `array_filter`. You never know how the array will be used later on. Or maybe clearer - all of that logic could be a loop.